### ISSUE TITLE
NÜWA: Visual Synthesis Pre-training for Neural visUal World creAtion

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#128]
![image](https://user-images.githubusercontent.com/46308407/145013581-d1a7ff7c-80cd-4c89-b5fa-fbb8d28f8038.png)

## TL; DR

- Replace vanilla transformer in VQ-GAN with 3D transformer.
- Represent image, video, text and sketch with 4D tensors (h, w, s, h) and uniformly model them. 
- Commonly applied to 8 tasks (T2I, T2V, I2I, TI2I, V2V, S2I, S2V, TV2V), among which 3 have recorded SOTA.

## Motivation

- Integrate multi-modal data into common representations and model them with strong 3D transformer.

## Previous Work

- VQ-VAE: Discretize continuous image manifold using vector-quantized codebook representations
- VQ-GAN: Replace pixel-level reconstruction loss with perceptual loss + GAN loss and train latent transformer in *next token prediction manner*.
- Dall-E: Pretrain dVAE (\sim VQ-VAE) with reconstruction task and train latent transformer for zero-shot T2I
- GODIVA: Auto-regressive T2V using VQ-VAE and transformer.

### Multimodal Representation

- Represent all data modalities with 4D tensor (h, w, s, d)
- Text: lower-case BPE
- Image: VQ-GAN codebook representation
- Video: Sequence of 2D VQ-GAN images
- Sketch: Separate VQ-GAN to embed (h,w,c) channeled one-hot segmentation map

### 3D Nearby Attention

- Similar to that used in SWIN-transformer.
- For image and video, apply 3x3 spatial window attention for computational efficiency.

### Training

- Pretrain on 3 tasks (T2I, Video prediction, T2V) using NLL objective
![image](https://user-images.githubusercontent.com/46308407/145014155-e91d4375-e3a1-4734-b4b4-ed52aa9fda61.png)

- As video prediction has no conditional text input, simply use "None" token.

### Datasets
- Text-to-Image: **Conceptual Captions** (256 x 256)
- Video Prediction: **Moments in Time**
- Text-to-Video: **VATEX** (336 x 336)

### Qualitative Evaulation
![image](https://user-images.githubusercontent.com/46308407/145014438-be08b531-d464-4a03-b3d6-38eb5d935c4d.png)
![image](https://user-images.githubusercontent.com/46308407/145014465-83411447-e784-4c85-bbad-c868a859cb05.png)
![image](https://user-images.githubusercontent.com/46308407/145014490-ae0482b3-f7bb-41a2-a25a-f4efb2130510.png)
![image](https://user-images.githubusercontent.com/46308407/145014504-4f1c2430-031a-46e6-bc54-943a16c16027.png)
![image](https://user-images.githubusercontent.com/46308407/145014525-d348af2a-74e8-43d5-a5e6-a4df42b66cde.png)
![image](https://user-images.githubusercontent.com/46308407/145014534-a8f44052-d969-4d9d-91a9-54bc0ba378aa.png)
![image](https://user-images.githubusercontent.com/46308407/145014597-870603b6-2fbe-47f7-b493-bbb2878314f4.png)

### More Results from Supple

![image](https://user-images.githubusercontent.com/46308407/145014775-031d75b6-b0b9-4e1c-938f-46afee5bf281.png)
![image](https://user-images.githubusercontent.com/46308407/145014792-e392eeb3-f4cb-4fd3-9a99-6ba3a545ae50.png)
![image](https://user-images.githubusercontent.com/46308407/145014811-d0c46fef-a00f-4633-8fda-c4e451c4d69c.png)
![image](https://user-images.githubusercontent.com/46308407/145014861-645c94a8-63b3-4c01-84c0-9e6c96e3ef61.png)
![image](https://user-images.githubusercontent.com/46308407/145014892-51df8da8-62fc-405a-962a-2c8747de3b2e.png)

### Quantitative Evaluation
![image](https://user-images.githubusercontent.com/46308407/145014665-505375d5-15e1-4fa3-9e42-95aea8dfa01c.png)
![image](https://user-images.githubusercontent.com/46308407/145014679-a0f90724-32ce-44cf-a4a7-5b5732c64390.png)
![image](https://user-images.githubusercontent.com/46308407/145014696-f0f5eb1e-0021-4855-9499-25b98aef49da.png)
![image](https://user-images.githubusercontent.com/46308407/145014705-7fb1f070-5086-4a70-8707-b8364d6955db.png)

## Comments

- Overall idea is very similar to that of VQ-GAN, and the novelty lies in the integration of multi-modalities. Plus the authors brag their abundance of computation resources. (64 A100 for 2 weeks... :+1: )
- Transformers are really shining in the era of multi-modalities. I guess now self-attention is the de facto operation for practically *all* tasks in deep learning research.
- The term modality heavily relies upon human perception system, i.e., how we the humans recognize the world. In essence, they are all just bits of *information*, which can be computed and understood using universal compute engines like the Transformer.
- There's going to be much excitement in multi-modality research, just that schools lack the resources to actively engage in.
